### DISCUSSION TITLE
Storage - custom adapter

### DISCUSSION THREAD

**Discussion by kwarpechowski** [directus/directus#14628] [Archive]
Hello,
In my case I must use custom storage for files. It is CDN hosted by external provider. I can sync only via JS SDK, http api,  rsync, ftp, scp or ssh.

Is there any possible write a custom storage connector and join it to directus?

I have another, fallback idea: to use stanard GCP/S3 connector and write auto replication to external provider or use local storage (upload dir) and sync this dir to provider. But, I think it is not clear architecture.

**Comment by rijkvanzanten** [comment]
There's currently no way to inject custom auth providers. See https://github.com/directus/directus/discussions/3107 for the duplicate feature request 👍🏻
